### PR TITLE
Remove deprecated `stochastic_weight_avg` argument from `Trainer`

### PR DESCRIPTION
**Pull request by rohitgr7**
## What does this PR do?

Part of #12521
required for: #11822

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda @carmocca @justusschock @kaushikb11 @awaelchli @rohitgr7 @tchaton

### CODE REVIEW

**Comment by rohitgr7** [comment]
@ananthsub how is this a breaking change?

**Comment by kaushikb11** [comment]
@rohitgr7 It would be a breaking change for the users as we are removing the Trainer flag.

File: CHANGELOG.md [hunk 1]
```diff
@@ -57,6 +57,9 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 ### Removed
 
+- Remove deprecated `stochastic_weight_averaging` flag from `Trainer` ([#12535](https://github.com/PyTorchLightning/pytorch-lightning/pull/12535))
```
Comment (ananthsub) [inline CHANGELOG.md:60]: let's match the entry right below
```suggestion
- Removed the deprecated `stochastic_weight_averaging` argument from the `Trainer` constructor ([#12535](https://github.com/PyTorchLightning/pytorch-lightning/pull/12535))
```

**Review by ananthsub — APPROVED** [review]

**Review by kaushikb11 — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -57,6 +57,9 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 ### Removed
 
+- Removed the deprecated `stochastic_weight_averaging` argument from the `Trainer` constructor ([#12535](https://github.com/PyTorchLightning/pytorch-lightning/pull/12535))
```
Comment (ananthsub) [inline CHANGELOG.md:60]:
```suggestion
- Removed the deprecated `stochastic_weight_avg` argument from the `Trainer` constructor ([#12535](https://github.com/PyTorchLightning/pytorch-lightning/pull/12535))
```

**Review by awaelchli — APPROVED** [review]

**Pull request merged**